flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStWriteBatchOperation.java

```diff
@@ -75,7 +74,10 @@ public CompletableFuture<Void> process() {
                             request.completeStateFuture();
                         }
                     } catch (Exception e) {
-                        throw new CompletionException("Error while adding data to ForStDB", e);
+                        for (ForStDBPutRequest<?, ?> request : batchRequest) {
```

Comment:
This will make the executor continue to run and may everytime `completeStateFutureExceptionally` all requests.
Could we just `completeStateFutureExceptionally` specific request ? Or fail the executor ?
WDYT ?

Reply:
Thanks for reminder @masteryhx , firstly this batch operation should indeed fail in this case.  Currently, the  AsyncFrameworkExceptionHandler will fail the job directly, then the StateExecutor will be destroy. I don't see the need to let the state executor continue executing, so I will fail the executor and update the pr.l